zerver/views/realm.py

```diff
@@ -60,6 +63,34 @@ def parse_jitsi_server_url(
 JITSI_SERVER_URL_MAX_LENGTH = 200
 
 
+check_group_setting: Validator[Union[int, Dict[str, List[int]]]] = check_union(
+    [
+        check_int,
+        check_dict_only(
+            [
+                ("direct_members", check_list(check_int)),
+                ("direct_subgroups", check_list(check_int)),
+            ]
+        ),
+    ]
+)
+
+
+def parse_group_setting_value(
+    setting_value: Union[int, Dict[str, List[int]]],
+) -> Union[int, AnonymousSettingGroupDict]:
+    if isinstance(setting_value, int):
+        return setting_value
+
+    if len(setting_value["direct_members"]) == 0 and len(setting_value["direct_subgroups"]) == 1:
+        return setting_value["direct_subgroups"][0]
+
+    return AnonymousSettingGroupDict(  # nocoverage
+        direct_members=setting_value["direct_members"],
```

Comment:
No setting that returns `AnonymousSettingGroupDict` currently in `views/realm`. Hence, `no coverage` for now